feature_engine/timeseries/forecasting/base_forecast.py

```diff
@@ -0,0 +1,214 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+from feature_engine.docstrings import (
+    Substitution,
+    _drop_original_docstring,
+    _feature_names_in_docstring,
+    _fit_not_learn_docstring,
+    _missing_values_docstring,
+    _n_features_in_docstring,
+)
+from feature_engine.validation import _return_tags
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+
+@Substitution(
+    missing_values=_missing_values_docstring,
+    drop_original=_drop_original_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    fit=_fit_not_learn_docstring,
+    n_features_in_=_n_features_in_docstring,
+)
+class BaseForecast(BaseEstimator, TransformerMixin):
```

Comment:
Minor point: I think naming this class BaseForecast could be misleading. It's not a base forecast and doesn't produce forecasts. The docstring is clear though. A more explicit but verbose name could be: `BaseForecastTransformer`, `BaseTimeseriesTransformer`.